snapshot/git/gitdb/local_data.go

```diff
@@ -19,42 +18,31 @@ type localValue struct {
 	kind valueKind
 }
 
-func (v *localValue) ID() snapshot.ID {
-	return snapshot.ID(fmt.Sprintf(localIDFmt, localIDText, v.kind, v.sha))
-}
-
-// parseID parses ID into a localValue
-func parseID(id snapshot.ID) (*localValue, error) {
-	if id == "" {
-		return nil, fmt.Errorf("empty snapshot ID")
-	}
-	parts := strings.Split(string(id), "-")
-	if len(parts) != 3 {
-		return nil, fmt.Errorf("did not scan 3 items from %s", id)
+func parseLocalID(id snapshot.ID, kind valueKind, parts []string) (*localValue, error) {
+	if len(parts) != 1 {
+		return nil, fmt.Errorf("cannot parse snapshot ID: expected 3 parts in local id %s", id)
```

Comment:
We expect three parts; really the parts handed in to parseLocalID are the additional parts (because parseID takes care of scheme and kind before calling parseLocalID).

Added a comment.